### PR TITLE
Big Bang OCI compose and Big Bang `--differential` fix

### LINKED ISSUE #1746: OCI Compose Extensions (BigBang)
### Is your feature request related to a problem? Please describe.
The current OCI skeleton feature focuses on Zarf primitives - we should also support extensions long term.

### Describe the solution you'd like

- **Given** I have a component that uses the big bang extension
- **When** I publish the package as a skeleton
- **Then** the valuesFiles are included in that skeleton

---

- **Given** I have an oci skeleton component that uses the big bang extension
- **When** I import the package as a skeleton and add merge overrides
- **Then** the local valuesFiles are merged with the skeleton's files

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This fixes an issue with BB not respecting differentials and adds the ability to OCI compose the BB extension.

## Related Issue

Fixes #1746 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: src/pkg/packager/compose.go [hunk 1]
```diff
@@ -256,6 +260,8 @@ func (p *Packager) fixComposedFilepaths(pathAncestry string, child types.ZarfCom
 		child.CosignKeyPath = composed
 	}
 
+	child = p.composeExtentions(pathAncestry, child)
```
Comment (Noxsios) [inline src/pkg/packager/compose.go:263]:
```suggestion
	child = p.composeExtensions(pathAncestry, child)
```

File: src/pkg/packager/extensions.go [hunk 2]
```diff
@@ -8,19 +8,78 @@ import (
...
-			return c, fmt.Errorf("unable to process bigbang extension: %w", err)
+	if component.Extensions.BigBang != nil {
+		component = bigbang.Compose(pathAncestry, component)
+	}
+
+	return component
+}
+
+// Check for any extensions in use and skeletonize their local files.
+func (p *Packager) skeletonizeExtentions() error {
```
Comment (Noxsios) [inline src/pkg/packager/extensions.go:58]:
```suggestion
func (p *Packager) skeletonizeExtensions() error {
```

File: src/pkg/packager/publish.go [hunk 3]
```diff
@@ -293,6 +293,11 @@ func (p *Packager) loadSkeleton() error {
 		return err
 	}
 
+	err = p.skeletonizeExtentions()
```
Comment (Noxsios) [inline src/pkg/packager/publish.go:296]:
```suggestion
	err = p.skeletonizeExtensions()
```

**Review by Noxsios — CHANGES_REQUESTED** [review]

File: src/pkg/packager/extensions.go [hunk 4]
```diff
@@ -8,19 +8,78 @@ import (
...
+
+	// Update the parent package config with the expanded sub components.
+	// This is important when the deploy package is created.
+	p.cfg.Pkg.Components = components
+
+	return nil
+}
+
+// Mutate any local files to be relative to the parent
+func (p *Packager) composeExtentions(pathAncestry string, component types.ZarfComponent) types.ZarfComponent {
```
Comment (Noxsios) [inline src/pkg/packager/extensions.go:46]:
```suggestion
func (p *Packager) composeExtensions(pathAncestry string, component types.ZarfComponent) types.ZarfComponent {
```

**Review by Noxsios — CHANGES_REQUESTED** [review]

File: src/pkg/packager/publish.go [hunk 5]
```diff
@@ -293,6 +293,11 @@ func (p *Packager) loadSkeleton() error {
 		return err
 	}
 
+	err = p.skeletonizeExtensions()
+	if err != nil {
+		return err
+  }
```
Comment (Noxsios) [inline src/pkg/packager/publish.go:299]: lint?

**Review by Noxsios — CHANGES_REQUESTED** [review]

**Review by YrrepNoj — APPROVED** [review]

File: src/pkg/packager/publish.go (hunk 5 above)
Comment (Racer159) [reply src/pkg/packager/publish.go]: Yeah that was strange - fixed now

**Review by Noxsios — APPROVED** [review]
lgtm

**Pull request merged**